### PR TITLE
enable providing a registry proxy via REGISTRY_URL

### PR DESCRIPTION
**Pull request by MaximilianMeister**
Signed-off-by: Maximilian Meister <mmeister@suse.de>

### CODE REVIEW

**Comment by MaximilianMeister** [comment]
this enables the ci in to set a different registry proxy to combat the slow downloading times NUE-PRV

File: process_manifest.rb [hunk 1]
```diff
@@ -6,13 +6,14 @@
 require "json"
 
 if ARGV.count != 3
-  puts "usage: #{$PROGRAM_NAME} container-manifests-dir velum-source-code-dir salt-dir"
+  puts "usage: #{$PROGRAM_NAME} container-manifests-dir velum-source-code-dir salt-dir [registry-url]"
```
Comment (ereslibre) [inline process_manifest.rb]: Why not use an envvar instead?

**Comment by kiall** [comment]
A small fix to this PR in https://github.com/kubic-project/caasp-devenv/pull/30

File: process_manifest.rb (hunk 1 above)
Comment (MaximilianMeister) [inline process_manifest.rb]: fixed

**Comment by MaximilianMeister** [comment]
> A small fix to this PR in #30

I even left out some more occurrences of `docker-testing-registry.suse.de` down the `process_manifest.rb` file. this is also fixed now.

File: start [hunk 2]
```diff
@@ -70,15 +75,15 @@ cp -r salt-admin-minion-config/* tmp/salt/config/admin
 for template in $(ls $CONTAINER_MANIFESTS_DIR/*.yaml)
 do
     log "Processing template $template..."
-    cat $template | ruby process_manifest.rb $CONTAINER_MANIFESTS_DIR $VELUM_DIR $SALT_DIR > manifests/$(basename $template)
+    cat $template | REGISTRY_URL=$REGISTRY_URL ruby process_manifest.rb $CONTAINER_MANIFESTS_DIR $VELUM_DIR $SALT_DIR > manifests/$(basename $template)
```
Comment (ereslibre) [inline start]: If `REGISTRY_URL` is exported there's no need to explicitly pass this as the environment is inherited, can you double check that?

File: start [hunk 3]
```diff
@@ -70,15 +75,15 @@ cp -r salt-admin-minion-config/* tmp/salt/config/admin
...
 done
 
 default_interface=$(awk '$2 == 00000000 { print $1 }' /proc/net/route)
 ip_address=$(ip addr show $default_interface | awk '$1 == "inet" {print $2}' | cut -f1 -d/)
 
 for template in $(ls manifest-templates/*.yaml)
 do
-    cat $template | ruby process_manifest.rb $CONTAINER_MANIFESTS_DIR $VELUM_DIR $SALT_DIR | sed -e "s#\${ip_address}#$ip_address#" > manifests/$(basename $template)
+    cat $template | REGISTRY_URL=$REGISTRY_URL ruby process_manifest.rb $CONTAINER_MANIFESTS_DIR $VELUM_DIR $SALT_DIR | sed -e "s#\${ip_address}#$ip_address#" > manifests/$(basename $template)
```
Comment (ereslibre) [inline start]: Same here.

**Comment by kiall** [comment]
http://jenkins.caasp.suse.net/job/caasp-devenv/job/PR-29/2/artifact/logs/docker-ps.log also shows this works, merging :)

File: start (hunk 2 above)
Comment (MaximilianMeister) [inline start]: it's not exported afaics, but that would have been an option too... to export it here https://github.com/kubic-project/caasp-devenv/pull/29/files#diff-ea2b2676c28c0db26d39331a336c6b92R54

but i prefered to have it explicit on the caller side

**Pull request merged**